stablehlo/reference/Element.cpp

```diff
@@ -601,6 +601,25 @@ Element atan2(const Element &e1, const Element &e2) {
                                      debugString(type).c_str()));
 }
 
+Element cbrt(const Element &el) {
+  auto type = el.getType();
+  if (isSupportedFloatType(type))
+    return Element(type, std::cbrt(el.getFloatValue().convertToDouble()));
+
+  if (isSupportedComplexType(type)) {
+    auto real = el.getComplexValue().real().convertToDouble();
+    auto imag = el.getComplexValue().imag().convertToDouble();
+    // cbrt(a + bi) = cbrt(sqrt(a^2 + b^2)) * (cos(1/3 * atan2(b, a)) + i *
+    // sin(1/3 * atan2(b, a)))
+    auto theta = std::atan2(imag, real) / 3.0;
+    return Element(
+        type, std::pow(std::pow(real, 2.0) + std::pow(imag, 2.0), 1.0 / 6.0) *
```

Comment:
Quote:
the n nth roots of z are given by
$r^{\frac {1}{n}}\left(\cos {\frac {x+2\pi k}{n}}+i\sin {\frac {x+2\pi k}{n}}\right)$

We assume n=3 (cube root) and k=0 as principal root:
$r^{\frac {1}{3}}\left(\cos {\frac {x}{3}}+i\sin {\frac {x}{3}}\right)$

Polar form:

$x=a+bi$

$r=\sqrt{a^{2}+b^{2}}$

$\theta_{a+bi}=atan2(b, a)$

Substituting variables:

$x^{\frac{1}{3}}=\left(\sqrt{a^{2}+b^{2}}\right)^{\frac{1}{3}}\left(\cos{\frac{1}{3}*atan2(b, a)}+i\sin{\frac{1}{3}*atan2(b,a)}\right)$

`1/6` power comes from doing both cubic and square roots. I've combined them to avoid losing more precision than necessary.

Reply:
Thanks!